crates/analyzer/src/traversal/expressions.rs

```diff
@@ -1767,7 +1767,21 @@ fn expr_call_builtin_value_method(
                         vec![],
                     );
                 }
-                Location::Storage { .. } => {}
+                Location::Storage { .. } => {
+                    if value_attrs.typ.is_base(context.db()) {
```

Comment:
I think we also have to handle the case where `typ == Type::Contract(_)` because it's a `newtype`'d address and is treated as a base type internally.

Reply:
Good catch! Fixed it.